lib/connections/service.go

```diff
@@ -36,6 +36,8 @@ var (
 	listeners = make(map[string]listenerFactory, 0)
 )
 
+const perDeviceWarningRate = 1.0 / (15 * 60) // Once per 15 minutes
```

Comment:
Suggest we should do this a bit more frequently? A version mis-match is almost never expected in a correct set-up: if it occurs, the user's going to be trying to find out why their devices won't connect. If the first message gets lost in some noise, we don't want them searching around for 15 mins before it pops up again, I would have thought..?